l10n/en/careers/404.ftl

```diff
@@ -0,0 +1,10 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+### URL: https://www-dev.allizom.org/careers/404/
+
+job-not-found-page-title = 404: Job Not Found
```

Comment:
Are we planning on localizing careers? If not, would it make more sense to avoid creating a new ftl file here?

Reply:
I asked internally and careers opportunities are currently only posted in English, so I'll go with removing the Fluent stuff altogether